core/modules/comment/css/comment_titles.css

```diff
@@ -0,0 +1,22 @@
+/* 
+ * @TODO - Remove this note
+ * Temporarily adding a new css file to deal with contrib themes
+ * that have already overwritten the primmary comments.css file.
+ * This strategy is still up for discussion.
+ */
+
+.hide_comment_title {
+  display: block;
+  /* Looking for a better idea here. */
+  text-indent: -99999px;
+  line-height: 0;
+}
+
+.hide_comment_title:after {
+  display: block;
+  /* Looking for a better/translatable idea here. */
+  content: "Permalink";
```

Comment:
As the constant name AND help text currently are - there shouldn't be visible text:

COMMENT_TITLE_HIDDEN and t('Comment titles are hidden.') rather indicate, that there's NO visible text at all.

Another suggestion: as the constants are new, we could name them in a less ambiguous way:

COMMENT_SUBJECT_HIDDEN - it's the subject, not actually the title.

Reply:
> As the constant name AND help text currently are - there shouldn't be visible text:

@indigoxela - Good points. The goal is to hide the title. BUT, I'm also trying to make sure we don't loose the permalink entirely.

My plan was to implement your idea that when the title is hidden, to replace it with a permalink icon. BUT, I don't know how to replace the title with an icon that is a link. So in it's current state, the "hidden" status actually replaces the title with the word "permalink." For me, this is an improvement, but not the end goal. Looking for better ideas or help replacing the title with an icon. 

> COMMENT_SUBJECT_HIDDEN - it's the subject, not actually the title.

I need guidance here, because I named the new setting "comment_subject_options" because it is more consistent with the old setting name. BUT, in the UI we don't use the word "subject" we use the term "title." 

1) I'm not sure if subject or title is more accurate and/or user friendly. I prefer title, in which case maybe I should drop "subject" completely OR the opposite.

Reply:
For now, I'm using "Title" instead of subject - because that is what has been used in the UI for a long time prior to this PR. Happy to discuss alternatives, but for now at least it's consistent.